stl/inc/limits

```diff
@@ -1013,20 +1015,83 @@ public:
 };
 
 #ifdef __cpp_lib_bitops // TRANSITION, VSO-1020212
+
+inline bool _Is_tzcnt_supported() {
+    // tzcnt and lzcnt support are indicated with
+    // different cpuid flags.
+    int cpuinfo[4];
+    __cpuid(cpuinfo, 0x07);
+    return cpuinfo[1] & (1 << 3);
+}
+
+
+// see "Hacker's Delight" section 5-3
+template <typename _Ty>
+constexpr int _Countl_zero_helper(_Ty _Val) {
+    _Ty _Yy = 0;
+
+    unsigned int _Nn = numeric_limits<_Ty>::digits;
+    unsigned int _Cc = numeric_limits<_Ty>::digits / 2;
+    do {
+        _Yy = static_cast<_Ty>(_Val >> _Cc);
+        if (_Yy != 0) {
+            _Nn  = _Nn - _Cc;
+            _Val = _Yy;
+        }
+        _Cc = _Cc >> 1;
+    } while (_Cc != 0);
+    return static_cast<int>(_Nn) - static_cast<int>(_Val);
+}
+
+// see "Hacker's Delight" section 5-4
+template <typename _Ty>
+constexpr int _Countr_zero_helper(_Ty _Val) {
+    return numeric_limits<_Ty>::digits

```

Comment:
I observe that if you extracted `_Digits`, you could avoid wrapping here - clearer code with the same number of lines.